test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractMintHTSSuite.java

```diff
@@ -605,9 +605,9 @@ private HapiApiSpec rollbackOnFailedAssociateAfterNonFungibleMint() {
 						getAccountBalance(TOKEN_TREASURY).hasTokenBalance(nonFungibleToken, 0),
 						childRecordsCheck(nestedMintTxn, CONTRACT_REVERT_EXECUTED,
 								recordWith()
-										.status(SUCCESS)
-										.newTotalSupply(1)
-										.serialNos(List.of(1L)),
+										.status(REVERTED_SUCCESS)
```

Comment:
What should happen if we have for example 2 child records, where the first one executes successfully and the second one fails. Should the first record have `SUCCESS` status and the second - `REVERTED_SUCCESS` or if a given child record fails, we should traverse through all of the previous child records and change their status to `REVERTED_SUCCESS`? 

I guess that for better traceability and user experience, it would be good that all successful child records to have `SUCCESS` status, but their changes not to be persisted in the state, as it's currently the case?

Reply:
The current nature of precompile allows the execution to continue even after a child transaction (precompile call fails). In that case, we can absolutely have a use-case in which we have a call to precompile that fails, the solidity code "catches" that error but decides to go forward and not interrupt the execution.

In that case, we will have:
1. Successful parent record (`SUCCESS`)
2. Successful child record (`SUCCESS`)
3. Reverted child record (`SOME_FAILURE_CODE`)

However, if the solidity code decides to "revert" the whole transaction due to the child failing, then we would have
1. Reverted parent (`CONTRACT_REVERT_EXECUTED`)
2. 1st child record with `REVERTED_SUCCESS`
3. 2nd child record with `SOME_FAILURE_CODE`

There is another case as-well, in which the 2nd child record **does not fail** but for some reason some other revert fails after the 2nd child record (precompile call) is being executed. In that case we will have:
1. Reverted parent (`CONTRACT_REVERT_EXECUTED`)
2. 1st child record with `REVERTED_SUCCESS`
3. 2nd child record with `REVERTED_SUCCESS`